tests/test_config.py

```diff
@@ -134,3 +136,30 @@ def test_french_defaults_and_override():
         cfg.behavioral_testing.neutral_token.prefix_list
         == config_defaults_per_language[SupportedLanguage.fr].prefix_list
     ), "Config did not take default French value for prefix list (neutral tokens)"
+
+
+def test_update_config(tiny_text_config, monkeypatch, dask_client):
+    # Changing config for a first time
+    partial_config = {"similarity": None}
+    new_config = update_config(tiny_text_config, partial_config)
+    assert not new_config.similarity
+    save_config(new_config)
+
+    with jsonlines.open(f"{new_config.artifact_path}/configs.jsonl", "r") as reader:
+        all_configs = [config for config in reader]
+    assert len(all_configs) == 1
+    loaded_config = all_configs[0]
+    assert loaded_config == new_config
+
+    # Changing config for a second time
+    partial_config_2 = {"dataset_warnings": {"min_num_per_class": 40}}
+    new_config_2 = update_config(new_config, partial_config_2)
+    assert new_config_2.dataset_warnings.min_num_per_class == 40
+    save_config(new_config_2)
+
+    with jsonlines.open(f"{new_config_2.artifact_path}/configs.jsonl", "r") as reader:
+        all_configs_2 = [config for config in reader]
+
+    assert len(all_configs_2) == 2
+    loaded_config_2 = all_configs_2[-1]
+    assert loaded_config_2 == new_config_2
```

Comment:
It seems we don't need all those `_2` variables and we can reuse the same variables. I find it makes it obvious for the reader that the second part of the test is independent from the first one.
```suggestion
    with jsonlines.open(f"{new_config.artifact_path}/configs.jsonl", "r") as reader:
        all_configs = list(reader)
    assert len(all_configs) == 1
    assert all_configs[0] == new_config

    # Changing config for a second time
    partial_config = {"dataset_warnings": {"min_num_per_class": 40}}
    new_config = update_config(new_config, partial_config)
    assert new_config.dataset_warnings.min_num_per_class == 40
    save_config(new_config)

    with jsonlines.open(f"{new_config.artifact_path}/configs.jsonl", "r") as reader:
        all_configs = list(reader)

    assert len(all_configs) == 2
    assert all_configs[-1] == new_config
```

Reply:
Ok! All other tests are like that though, I thought it was always best to not reuse variables, so that is good to know!

Reply:
[Here](https://github.com/ServiceNow/azimuth/pull/349/files#diff-a6ba4243c32b086df6de7b529feda770ffbdb35b39c4f400a354945e363144edR122-R131) for example, you already reuse the same `loaded_configs` variable.